spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -114,11 +139,65 @@ public void onSuccess(String result) {
 
 	@Override
 	public Subscriber subscribe(String subscription, MessageReceiver messageHandler) {
-		Subscriber subscriber = this.subscriberFactory.getSubscriber(subscription, messageHandler);
+		Subscriber subscriber = this.subscriberFactory.createSubscriber(subscription, messageHandler);
 		subscriber.startAsync();
 		return subscriber;
 	}
 
+	/**
+	 * Pulls messages synchronously, on demand, using the pull request in argument.
+	 *
+	 * <p>
+	 * This method acknowledges all received messages.
+	 * @param pullRequest pull request containing the subscription name
+	 * @return the list of {@link PubsubMessage} containing the headers and payload
+	 */
+	private List<PubsubMessage> pull(PullRequest pullRequest) {
+		Assert.notNull(pullRequest, "The pull request cannot be null.");
+		Assert.notNull(this.subscriberFactory.createSubscriberStub(),
+				"The SubscriberStub cannot be null.");
+
+		try {
+			SubscriberStub subscriber = this.subscriberFactory.createSubscriberStub();
```

Comment:
TODO: I think we actually need to `close()` this subscriber after we're done.

Reply:
Isn't that what we can do with the `try-with-resources`: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html?
```
try(SubscriberStub subscriber = this.subscriberFactory.createSubscriberStub()) {
   ...
}
```

Reply:
Yep. Thanks!